experimental/oak_async/src/io.rs

```diff
@@ -72,6 +73,32 @@ impl<T: Decodable> Drop for ChannelRead<T> {
     }
 }
 
+/// `Stream` representing a sequence of asynchronous reads from a channel.
+pub struct ChannelReadStream<T: Decodable>(
+    // Note: `Stream` could be implemented directly on the `ChannelRead` type. Unfortunately the
```

Comment:
```suggestion
    // Note: `Stream` could not be implemented directly on the `ChannelRead` type. Unfortunately the
```

Reply:
Aah, the phrasing here is a bit confusing. This is possible, it would just be very un-ergonomic to use. I will try to explain that a bit better.